crypto/rand/rand_lib.c

```diff
@@ -32,15 +108,51 @@
 # include "internal/e_os.h"
 # include "internal/property.h"
 
+/*
+ * The default name for the random provider.
+ * This ensures that the FIPS provider will supply libcrypto's random byte
+ * requirements.
+ */
+static const char random_provider_fips_name[] = "fips";
+
+/*
+ * Utility functions to help handling the default random provider name.
+ * A pointer compare is not sufficient to detect a name being unmodified
+ * because a loaded provider can have its own copy of the global on some
+ * platforms (MIPS).  We therefore resort to string comparisons.
+ */
+static void free_random_provider_name(RAND_GLOBAL *dgbl)
+{
+    if (dgbl->random_provider_name != NULL
+            && strcmp(dgbl->random_provider_name,
+                      random_provider_fips_name) != 0)
```

Comment:
Why strcmp and not just pointer comparison?

Reply:
On some MIPS implementations, the `static const` data is replicated in different places.
Yes, it's wrong and not adhering to the standards but that's what happens.
You wouldn't believe how long this took to figure out....

Reply:
Oh, wow.